File: java/step3-method-sep/src/test/java/badcode/MassiveRaceTest.java
package badcode;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

class MassiveRaceTest {

    @DisplayName("2라운드 경주에서 가장 많이 전진한 자동차가 우승한다")
    @Test
    void tA() {
        MassiveRace r = new MassiveRace();
        int[] rv = { 4, 3, 4, 4, 3, 4 };
        String result = r.race("kim,lee,park", 2, rv);
        assertThat(result).isEqualTo("kim, park가 최종 우승했습니다.");
    }

    @DisplayName("모든 자동차가 같은 거리를 이동하면 전원 우승한다")
    @Test
    void tB() {
        MassiveRace r = new MassiveRace();
        int[] rv = { 4, 4, 4 };
        String result = r.race("kim,lee,park", 1, rv);
        assertThat(result).isEqualTo("kim, lee, park가 최종 우승했습니다.");
    }

    @DisplayName("자동차가 1대뿐이면 예외가 발생한다")
    @Test
    void tC() {
        MassiveRace r = new MassiveRace();
        assertThatThrownBy(() -> r.race("kim", 1, new int[] { 4 }))
                .isInstanceOf(IllegalArgumentException.class);
    }

    @DisplayName("자동차 이름이 5자를 초과하면 예외가 발생한다")
    @Test
    void tD() {
        MassiveRace r = new MassiveRace();
        assertThatThrownBy(() -> r.race("kim,abcdef", 1, new int[] { 4, 4 }))
                .isInstanceOf(IllegalArgumentException.class);
    }
}

File: ts/step3-method-sep/massiveRace.test.ts
import { MassiveRace } from './massiveRace';

describe('MassiveRace', () => {
    it('테스트A', () => {
        const r = new MassiveRace();
        const result = r.race('kim,lee,park', 2, [4, 3, 4, 4, 3, 4]);
        expect(result).toBe('kim, park가 최종 우승했습니다.');
    });

    it('테스트B', () => {
        const r = new MassiveRace();
        const result = r.race('kim,lee,park', 1, [4, 4, 4]);
        expect(result).toBe('kim, lee, park가 최종 우승했습니다.');
    });

    it('테스트C', () => {
        const r = new MassiveRace();
        expect(() => r.race('kim', 1, [4])).toThrow();
    });

    it('테스트D', () => {
        const r = new MassiveRace();
        expect(() => r.race('kim,abcdef', 1, [4, 4])).toThrow();
    });
});
